### ISSUE TITLE
Client error handling?

### ISSUE THREAD

**Issue by nyarly** [tenderlove/hana#7]
Per 6901, JSON pointer error handling is up to the implementation. I see that Pointer is used here primarily to implement Patch, but would it be possible to make it so that an error could be handled by the user of Hana?  

Specifically, in cases where there's no value at (an intermediate) index there's a lot of room to do useful things like fill in empty objects.

If so, I'll volunteer a PR.

**Comment by tenderlove** [comment]
Yes, I think this is fine.  Can you send a PR with a suggested implementation?  I'm not sure what you have in mind exactly, so I'd like to see the code! :)
